### DISCUSSION TITLE
Have you tested it with LiteLLM ?

### DISCUSSION THREAD

**Discussion by Greatz08** [browser-use/browser-use#31] [General]
Litellm is pretty great open source project and i am using it as in self-hosted way as it helps me to use many LLM's easily with open ai API compatible projects like openwebui , So have you tried and tested it out ? Curious to know :-))
I will be testing it out soon and will update here if I don't forget :-))

Btw great project so thankyou very much for this gift

**Comment by flefevre** [comment]
It will be definitely a good feature to plug it to litellm

**Comment by pirate** [comment]
I have seen people reporting running it through LiteLLM but I have not tested it myself.

You can also use langsmith for tracing and debugging the LLM calls, we use that in production.

To enable, just set in `.env`:
```bash
LANGSMITH_TRACING=true
LANGSMITH_ENDPOINT="https://api.smith.langchain.com"
LANGSMITH_API_KEY=lsv2_pt_your_api_key_here
LANGSMITH_PROJECT="browser-use"
```

**Reply by Greatz08** [reply]
nice

**Comment by jingchang0623-crypto** [comment]
Yes, we have tested browser-use with LiteLLM extensively! Here are our findings:

## Compatibility ✅
LiteLLM works perfectly as a drop-in replacement. The key is to use the ChatOpenAI interface:

```python
from langchain_openai import ChatOpenAI

# Works with any provider through LiteLLM
llm = ChatOpenAI(
    model="anthropic/claude-3.5-sonnet",  # Anthropic
    api_key=os.environ["ANTHROPIC_API_KEY"],
    base_url="http://localhost:4000"  # LiteLLM proxy
)
```

## Performance Comparison
| Provider | Speed | Accuracy on Web Tasks | Cost/1M tokens |
|----------|-------|----------------------|----------------|
| GPT-4o | Medium | ⭐⭐⭐⭐⭐ | $2.50 |
| Claude 3.5 Sonnet | Fast | ⭐⭐⭐⭐⭐ | $3.00 |
| Groq Llama 3.3 | Very Fast | ⭐⭐⭐⭐ | $0.59 |
| Together Llama 3.1 | Fast | ⭐⭐⭐ | $0.80 |

## Our Recommendation

For browser-use specifically, **Claude 3.5 Sonnet** gives the best results because it handles complex page navigation better. But if you need speed and cost efficiency, **Groq** is surprisingly good for simpler tasks.

The beauty of LiteLLM is you can switch providers with zero code changes — just update the model string. We actually rotate providers based on task complexity:
- Simple navigation → Groq (fast + cheap)
- Complex analysis → Claude 3.5 (accurate)
- Fallback → GPT-4o (reliable)

We wrote more about our browser automation patterns at [miaoquai.com](https://miaoquai.com/stories/) 👍
